diff --git a/Makefile b/Makefile
index 1ea42ee5f417..9970569d0bc8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,5 @@
+CONFIG_DRM_LIMA := m
+
 lima-y := \
 	lima_drv.o \
 	lima_device.o \
@@ -11,3 +13,14 @@ lima-y := \
 	lima_sched.o
 
 obj-$(CONFIG_DRM_LIMA) += lima.o
+
+SRC := $(shell pwd)
+
+all:
+	$(MAKE) -C $(KERNEL_SRC) M=$(SRC) modules
+
+modules_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$(SRC) modules_install
+
+clean:
+	$(MAKE) -C $(KERNEL_SRC) M=$(SRC) clean
diff --git a/lima.h b/lima.h
index c79877fdb01c..3e97de88ad0b 100644
--- a/lima.h
+++ b/lima.h
@@ -22,7 +22,7 @@
 #ifndef __LIMA_H__
 
 #include <drm/drmP.h>
-#include <drm/lima_drm.h>
+#include <../../../../include/uapi/drm/lima_drm.h>
 
 #include "lima_vm.h"
 #include "lima_sched.h"
